### PR TITLE
Updated build scripts and README for live reload

### PR DESCRIPTION
**Pull request by tjaffri**
Looks like a big change but basically all I did was:
1. Create a fresh ionic project using the latest ionic beta tools
2. Copied the build scripts (gulpfile, etc) into our project. Some deprecated files were deleted.
3. Updated README with updated instructions

This fixes live reload, but note the changes in the README. Specifically you don't need to do "gulp build" anymore... that is taken care of automatically when you do "ionic serve"

You do need to install an updated ionic CLI, and that is updated in the README as well.

### CODE REVIEW

**Comment by tjaffri** [comment]
Oh, this resolves #16

**Comment by tjaffri** [comment]
@arjun-msft since you cared about this, perhaps you could review and merge?

Caution: my windows dev environment is hosed so I tested only on Mac. That has caused angst in the past so please verify this works on Windows before merging. Since I just copy-pasted from the standard template I don't expect there to be many issues.

**Comment by tjaffri** [comment]
I should note that this seems to work only if you change html/css. If you change typescript you still need to do "ionic serve" again (which does a rebuild).

For simple html/css changes this is still useful (and I guess default behavior). We might be able to do some more work to extend this to also live-reload if typescript changes... in the future.

File: README.md [hunk 1]
```diff
@@ -12,10 +12,11 @@ Get your dev environment set up (PC or Mac):
 * [Install Node](https://nodejs.org/en/download/)
 * Choose your favorite IDE, e.g. [Visual Studio Code](https://code.visualstudio.com/).
 
-Install the Ionic Framework, Cordova and Typings using npm. Note that we are using ionic 2, which is currently in beta.
+Install the Ionic Framework and Cordova using npm. Note that we are using ionic 2, which is currently in beta.
 
 ```bash
-$ npm install -g ionic@2.0.0-beta.10 cordova@6.2.0 gulp@3.9.1 tslint@3.13.0 typescript@1.8.10
```
Comment (arjun-msft) [inline README.md:18]: did you intentionally remove the gulp/tslint/typscript references?

File: typings/index.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+/// <reference path="globals/es6-shim/index.d.ts" />
```
Comment (arjun-msft) [inline typings/index.d.ts:1]: is this file necessary? this just looks like a comment

File: typings/browser/ambient/es6-shim/es6-shim.d.ts [hunk 3]
```diff
@@ -1,670 +0,0 @@
-// Compiled using typings@0.6.8
```
Comment (arjun-msft) [inline typings/browser/ambient/es6-shim/es6-shim.d.ts:1]: why do we not need this file anymore?

File: gulpfile.js [hunk 4]
```diff
@@ -1,147 +1,74 @@
...
+var tslint = require('ionic-gulp-tslint');
+
+var isRelease = argv.indexOf('--release') > -1;
+
+gulp.task('watch', ['clean'], function(done){
+  runSequence(
+    ['sass', 'html', 'fonts', 'scripts'],
+    function(){
+      gulpWatch('app/**/*.scss', function(){ gulp.start('sass'); });
+      gulpWatch('app/**/*.html', function(){ gulp.start('html'); });
```
Comment (arjun-msft) [inline gulpfile.js:44]: in my local version I had to add *.ts as well so that it would watch changes to the ts files. How do changes in ts files trigger a new build?

File: typings/index.d.ts (hunk 2 above)
Comment (tjaffri) [inline typings/index.d.ts:1]: This is the "bundle file" that describes all the typings in this project (currently just this one). Used by IDEs to resolve "intellisense"... not compiled that's why it looks like a comment. See https://github.com/typings/typings/blob/master/docs/faq.md (the module resolutions section).

Also, if it helps this file is created by default if you do a clean "ionic start myApp --v2 --ts".

File: gulpfile.js (hunk 4 above)
Comment (tjaffri) [inline gulpfile.js:44]: Magic :-) 

The ionic guys introduced this new thing and got rid of webpack: https://www.npmjs.com/package/ionic-gulp-browserify-typescript.

This new module is doing the watch (line 45 is where that option is set). I have tried this... when you change a ts file during ionic serve it detects the change and live reloads.

Oh, and if it helps this gulpfile is copied straight from a fresh ""ionic start myApp --v2 --ts".

File: typings/browser/ambient/es6-shim/es6-shim.d.ts (hunk 3 above)
Comment (tjaffri) [inline typings/browser/ambient/es6-shim/es6-shim.d.ts:1]: I think the typings package for es6-shim (and the format of typings in general) has been updated to no longer use "ambient" and instead use "global" terminology. That's why this file has now been moved to the global folder and renamed to index.d.ts.

I couldn't find a good reference for when this changed, but I did confirm that if I create a fresh app via "ionic start myApp --v2 --ts" then this file is removed and the typings/global/es6-shim/ folder is created (with the index.d.ts and typings.json)

File: README.md (hunk 1 above)
Comment (tjaffri) [inline README.md:18]: Yes, I think those should get installed as dependencies of the packages in package.json. I was just copying what the latest ionic2 templates have in their build process and they only talk about installing cordova and ionic globally.
